src/blocks/recommendations/components/parsely-recommendations-fetcher.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { useDebounce } from '@wordpress/compose';
+import { useCallback, useEffect } from '@wordpress/element';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import { setError, setRecommendations } from '../actions';
+import { useRecommendationsStore } from '../recommendations-store';
+
+const ParselyRecommendationsFetcher = ( { boost, limit, sort } ) => {
+	const {	dispatch } = useRecommendationsStore();
+
+	const query = {
+		boost,
+		limit,
+		sort,
+		url: window.location.href,
+	};
+
+	async function fetchRecommendationsFromWpApi() {
+		return apiFetch( {
+			path: addQueryArgs( '/wp-parsely/v1/related', { query } ),
+		} );
+	}
+
+	async function fetchRecommendations() {
+		let response;
+		let errorMessage;
+
+		try {
+			response = await fetchRecommendationsFromWpApi();
+		} catch ( wpError ) {
+			errorMessage = wpError;
+		}
+
+		if ( response?.error ) {
+			errorMessage = response.error;
+		}
+
+		if ( errorMessage ) {
+			dispatch( setError( { error: response.error } ) );
+			return;
+		}
+
+		const data = response?.data || [];
+		dispatch( setRecommendations( { recommendations: data } ) );
+	}
+
+	const reinvocationDelay = 300; // Reinvoke function at most every this amount of milliseconds.
```

Comment:
Could we move this out of the function, at the top of the file? We don't need to redeclare each time we run it.